frontend/src/metabase/css/core/colors.module.css

```diff
@@ -1,9 +1,24 @@
 /* NOTE: DO NOT ADD COLORS WITHOUT EXTREMELY GOOD REASON AND DESIGN REVIEW
  * NOTE: KEEP SYNCRONIZED WITH:
  * frontend/src/metabase/ui/utils/colors.ts
+ * frontend/src/metabase/lib/colors/palette.ts
  * frontend/src/metabase/styled-components/containers/GlobalStyles/GlobalStyles.tsx
  * frontend/src/metabase/styled-components/theme/css-variables.ts
  */
+
+/* This file contains two types of colors
+ *
+ * 1. Semantic colors
+ * 2. Base colors
+ *
+ * Semantic colors are the colors that are used in the application. They are derived
+ * from the base colors.
+ *
+ * Base colors are the colors that should never change and never be used outside this file,
+ * or in any components or CSS properties. They should only be used to define the semantic
+ * colors in this file.
```

Comment:
I think there are still cases for base colors to be defined outside this file.
1. ad-hoc colors: Like colors that are one-off, we could potentially create one-off semantic colors that are based on base colors e.g. I did this for [SVG icons](https://github.com/metabase/metabase/blob/64f130cc7f090918c20c0d242d7d8660b9c1f0c7/frontend/src/metabase/admin/settings/components/widgets/EmbeddingOption/EmbeddingOption.module.css) with different color schemes
2. Theming: We want to be able to override these semantic colors elsewhere, for example
    https://github.com/metabase/metabase/blob/17cefb94c533492a7e6bb9fef312d222798ea24a/frontend/src/metabase/public/components/EmbedFrame/EmbedFrame.module.css#L59-L68
    Or do you think such overrides should live within this file?

Reply:
🤔

Reply:
resolving. I also [agree that we should declare all semantic colors inside `colors.module.css`](https://metaboat.slack.com/archives/C057WD5L0JG/p1730732847564649?thread_ts=1730317945.468619&cid=C057WD5L0JG)